cli/pcluster/config/mappings.py

```diff
@@ -655,40 +656,246 @@
+    }),
+    ("s3_read_write_resource", {
+        "cfn_param_mapping": "S3ReadWriteResource",
+        "update_policy": UpdatePolicy.SUPPORTED
+    }),
+    # Customization
+    ("template_url", {
+        # TODO add regex
+        "validators": [url_validator],
+        # Ignored during update since we force using previous template
+        "update_policy": UpdatePolicy.IGNORED
+    }),
+    ("shared_dir", {
+        "type": SharedDirCfnParam,
+        "allowed_values": ALLOWED_VALUES["file_path"],
+        "cfn_param_mapping": "SharedDir",
+        "default": "/shared",
+        "validators": [shared_dir_validator],
+        "update_policy": UpdatePolicy.UNSUPPORTED
+    }),
+    ("enable_efa", {
+        "allowed_values": ["compute"],
+        "cfn_param_mapping": "EFA",
+        "validators": [efa_validator],
+        "update_policy": UpdatePolicy.UNSUPPORTED
+    }),
+    ("ephemeral_dir", {
+        "allowed_values": ALLOWED_VALUES["file_path"],
+        "default": "/scratch",
+        "cfn_param_mapping": "EphemeralDir",
+        "update_policy": UpdatePolicy.UNSUPPORTED,
+    }),
+    ("encrypted_ephemeral", {
+        "default": False,
+        "type": BoolCfnParam,
+        "cfn_param_mapping": "EncryptedEphemeral",
+        "update_policy": UpdatePolicy.UNSUPPORTED,
+    }),
+    ("custom_ami", {
+        "cfn_param_mapping": "CustomAMI",
+        "allowed_values": ALLOWED_VALUES["ami_id"],
+        "validators": [ec2_ami_validator],
+        "update_policy": UpdatePolicy.UNSUPPORTED,
+    }),
+    ("pre_install", {
+        "cfn_param_mapping": "PreInstallScript",
+        # TODO add regex
+        "validators": [url_validator],
+        "update_policy": UpdatePolicy.COMPUTE_FLEET_STOP,
+    }),
+    ("pre_install_args", {
+        "cfn_param_mapping": "PreInstallArgs",
+        "update_policy": UpdatePolicy.COMPUTE_FLEET_STOP,
+    }),
+    ("post_install", {
+        "cfn_param_mapping": "PostInstallScript",
+        # TODO add regex
+        "validators": [url_validator],
+        "update_policy": UpdatePolicy.COMPUTE_FLEET_STOP,
+    }),
+    ("post_install_args", {
+        "cfn_param_mapping": "PostInstallArgs",
+        "update_policy": UpdatePolicy.COMPUTE_FLEET_STOP,
+    }),
+    ("extra_json", {
+        "type": ExtraJsonCfnParam,
+        "cfn_param_mapping": "ExtraJson",
+        "update_policy": UpdatePolicy.COMPUTE_FLEET_STOP,
+    }),
+    ("additional_cfn_template", {
+        "cfn_param_mapping": "AdditionalCfnTemplate",
+        # TODO add regex
+        "validators": [url_validator],
+        "update_policy": UpdatePolicy.UNSUPPORTED,
+    }),
+    ("tags", {
+        "type": JsonCfnParam,
+        "update_policy": UpdatePolicy.UNSUPPORTED,
+    }),
+    ("custom_chef_cookbook", {
+        "cfn_param_mapping": "CustomChefCookbook",
+        # TODO add regex
+        "validators": [url_validator],
+        "update_policy": UpdatePolicy.UNSUPPORTED,
+    }),
+    ("custom_awsbatch_template_url", {
```

Comment:
We can move it in the SIT one (let's double check if we're skipping the installation of the cli in non-awsbatch cases, I think so)

Reply:
Done